system/jhtree/ctfile.cpp

```diff
@@ -78,11 +83,15 @@ void SwapBigEndian(KeyHdr &hdr)
     _WINREV(hdr.fileSize);
     _WINREV(hdr.nodeKeyLength);
     _WINREV(hdr.version);
+    _WINREV(hdr.keyFlags);
     _WINREV(hdr.blobHead);
     _WINREV(hdr.metadataHead);
     _WINREV(hdr.bloomHead);
     _WINREV(hdr.partitionFieldMask);
     _WINREV(hdr.firstLeaf);
+    //Clone the old flags byte into the new flags word for backwards compatibility
+    if (read)
+        hdr.keyFlags |= (byte)hdr.oldktype;
```

Comment:
Not convinced this works - won't it be 0xffff in an old key.

Also rather unexpected to find this code in a function called swapBigEndian... 

A comment on the field definition that it gets updated on load to account foc compatibility issues would be helpful